### PR TITLE
Implement reduce method

### PR DESCRIPTION
**Pull request by a-kotova**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Chapter {
+  reduce(array, callback, initialValue = null) {
```
Comment (iharzh) [inline index.js]: Please provide some error message in case there is no init value

**Review by iharzh — CHANGES_REQUESTED** [review]

File: index.js [hunk 2]
```diff
@@ -1,4 +1,17 @@
 class Chapter {
+  reduce(array, callback, initialValue) {
+    try {
```
Comment (iharzh) [inline index.js]: `try..catch` block is unnecessary here

File: index.js [hunk 3]
```diff
@@ -1,4 +1,17 @@
...
+      if (initialValue === undefined) throw new Error();
```
Comment (iharzh) [inline index.js]: you can move the error text from `catch` block right inside this error

**Review by iharzh — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (a-kotova) [reply index.js]: Removed

**Review by iharzh — APPROVED** [review]

**Pull request merged**